File: programmers/level1/크레인 인형뽑기 게임.py
import collections
def solution(board, moves):
    st_board = [collections.deque() for i in range(len(board))]
    stack = []
    for i in range(len(board)):
        for j in range(len(board[i])):
            if board[i][j]: st_board[j].appendleft(board[i][j])
    box = []
    cnt = 0
    print(st_board)
    for mv in moves:
        if len(st_board[mv-1])>0:
            box.append(st_board[mv-1].pop())
        
        if len(box)>1 and box[-1]==box[-2]:
            cnt += 2
            box.pop()
            box.pop()
    return cnt

File: programmers/level1/크레인 인형뽑기 게임.java
import java.util.Stack;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> st = new Stack<>();
        for(int move: moves) {
            for(int h=0;h<board.length;h++) {
                if(st.empty() && board[h][move-1] != 0) { //스택이 비었으면 넣기
                    st.push(board[h][move-1]);
                    board[h][move-1] = 0;
                    break;
                } else if(!st.empty() && board[h][move-1] == st.peek()) { //스택에 인형 있고, 지금 찾은 것과 같으면 스택에서 제거
                    board[h][move-1] = 0;
                    st.pop();
                    answer += 2;
                    break;
                } else if(board[h][move-1] != 0) { //인형 뽑음
                    st.push(board[h][move-1]);
                    board[h][move-1] = 0;
                    break;
                }
            }
        }
        return answer;
    }
}